regex-syntax/src/parser.rs

```diff
@@ -2375,6 +2384,46 @@ mod tests {
     }
 
     #[test]
+    fn ignore_space_escape_unicode_name() {
+        assert_eq!(p(r"(?x)\ p { Y i }"), Expr::Class(class(YI)));
+    }
+
+    #[test]
+    fn ignore_space_escape_octal() {
+        assert_eq!(p(r"(?x)\ 1 2 3"), lit('S'));
```

Comment:
Seems a bit weird that it's allowed to add space between digits of a number, but that seems to be the closest to the current behavior.